### ISSUE TITLE
Bump flot from 0.8.3 to 4.2.1

### ISSUE THREAD

**Issue by jdlrobson** [internetarchive/openlibrary#4396]
The project uses an outdated version of the flot library. We should upgrade it, making the required changes to make our code work with the new version.

### Steps to Reproduce
1. Pull down the patch https://github.com/internetarchive/openlibrary/pull/4346
2. Run `npm run build-assets:js` and see a JS error `ModuleNotFoundError: Module not found: Error: Can't resolve 'flot/jquery.flot.stack.js' in '/openlibrary/openlibrary/plugins/openlibrary/js/graphs'`
3. Make changes to code to avoid this JS error.
4. Confirm the graphs on http://localhost:8080/subjects/fantasy#sort=date_published&ebooks=true and  http://localhost:8080/publishers/Barnes_&_Noble should continue to render.

### Stakeholders
@jdlrobson

**Comment by cclauss** [comment]
@jdlrobson @jamesachamp This is the continuation of #4011...  What are the next steps?

**Comment by jdlrobson** [comment]
@cclauss someone needs to analyse what this entails.

I guess we'd need to find some upgrade notes for jumping from 0.8.3 to 4.2.1 given that's several major versions. We're going to need to change the underlying JS that uses the library.

If not trivial, we may want to take the opportunity to reconsider our graphing library as part of this.

**Comment by RayBB** [comment]
I don't think this is a high priority. I think we're more likely that @lokesh migrates us to a modern lit component or similar.

As such, closing this.
